### ISSUE TITLE
[BUG] "watch" files

### ISSUE THREAD

**Issue by pwilczynskiclearcode** [terramate-io/terramate#1780]
**Describe the bug**
Either `watch = []` option doesn't work or I don't understand it.
Providing whatever non-existing paths `watch = ["/non-existing", "/a-file-is-not-there", "./relative-non-existing"]` doesn't raise any [validation error](https://github.com/terramate-io/terramate/blob/main/config/stack.go#L253-L279) when running `terramate --changed list` I haven't manage to provide any file which changes would be detected by `watch=`.

https://terramate.io/docs/cli/stacks/configuration#watch

**Expected behavior**
* Changing a file that's listed on `watch = []` list to be resulting in `terramate --changed list` printing the stack
* Providing a file out of terramate root dir in `watch = []` to actually raise the exception/warning
* Documentation to be more descriptive on how to use and what to expect. Could the paths be project-relative or stack-relative or only absolute?

**Environment (please complete the following information):**
 - OS: Linux
 - OS Version: Ubuntu 24.04
 - Git Version: git version 2.43.0
 - Terramate Version: `0.9.0`

**Comment by i4ki** [comment]
Hi @pwilczynskiclearcode 

Did you commit the changed files (the watched file)? Terramate change detection still depend on git for figuring repo changes and then backtrack changes to specific stacks if they are watched.

The watch validation does not care if the file does not exist, similarly to `after`, `before`, `wants` and `wanted_by`, but I agree that we must at least give a warning.

Please let me know if committing works, because if not, then indeed we have a bug.
I tested the basic cases here now and it seems to be working fine.

**Comment by soerenmartius** [comment]
> The watch validation does not care if the file does not exist, similarly to after, before, wants and wanted_by, but I agree that we must at least give a warning.


I agree we should add warnings here @i4ki

**Comment by soerenmartius** [comment]
> Hi @pwilczynskiclearcode
> 
> Did you commit the changed files (the watched file)? Terramate change detection still depend on git for figuring repo changes and then backtrack changes to specific stacks if they are watched.
> 
> The watch validation does not care if the file does not exist, similarly to `after`, `before`, `wants` and `wanted_by`, but I agree that we must at least give a warning.
> 
> Please let me know if committing works, because if not, then indeed we have a bug. I tested the basic cases here now and it seems to be working fine.

@pwilczynskiclearcode I just tested the `watch` attribute and couldn't reproduce the issue so far. @pwilczynskiclearcode did you as mentioned by @i4ki commit the file?

**Comment by pwilczynskiclearcode** [comment]
Yes. It works like it suppose to when committing the changes. It could be just better documented :)
